src/model.py

```diff
@@ -342,6 +349,11 @@ async def generate(self, request):
         Forwards single request to LLM engine and returns responses.
         """
         response_sender = request.get_response_sender()
+        response_state = [
```

Comment:
maybe we can use dictionary instead of a list? this way in the code we can be more clear, i.e. instead of `response_state[1], we will use `response_state["is_cancelled"]`, or have a ResponseState class

Reply:
yes, a dictionary is cleaner - let me do a quick check if there is any perf regression.

Reply:
there is no regression, let's use the dictionary variant